components/user-store/org.wso2.carbon.user.cassandra/src/main/java/org/wso2/carbon/user/cassandra/Util.java

```diff
@@ -35,18 +35,21 @@
 import org.wso2.carbon.user.core.multiplecredentials.Credential;
 
 import java.nio.charset.Charset;
+import java.security.SecureRandom;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.util.Random;
 
 public class Util {
 
-    protected static Random random = new Random();
+    private static final String SHA_1_PRNG = "SHA1PRNG";
+    protected static Random secureRandom;
```

Comment:
No need to define a class variable here. Instead use a local variable within the static method since you always instantiate